netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -295,9 +309,20 @@ public Attributes getAttributes() {
 
   @Override
   public Future<TransportTracer.Stats> getTransportStats() {
-    SettableFuture<TransportTracer.Stats> ret = SettableFuture.create();
-    ret.set(null);
-    return ret;
+    if (channel.eventLoop().inEventLoop()) {
+      // This is necessary, otherwise we will block forever if we get the future from inside
```

Comment:
This comment seems incorrect since the swap to Future.

Reply:
The comment is referring to how this workaround lets us do Future.get() on the return value while on the event loop. I suppose we can attach a listener to the Future to work around this but it may not be very obvious why the Future blocks forever.